### PR TITLE
Refactor multiple choice

### PR DESCRIPTION
**Pull request by nick-benoit14**

### CODE REVIEW

File: client/js/components/assessments/item.jsx [hunk 1]
```diff
@@ -9,12 +9,14 @@ export default class Item extends React.Component{
 
   static propTypes = {
     question                   : React.PropTypes.object.isRequired,
+    response                   : React.PropTypes.array.isRequired,
```
Comment (fugue88) [inline client/js/components/assessments/item.jsx]: Let's start documenting these.  Let's try an inline comment at the end of the line for now.  Also, close up the spacing some, but keep things aligned.

File: client/js/components/common/radio_button.jsx [hunk 2]
```diff
@@ -2,22 +2,27 @@
...
-import Styles                 from "../../themes/selection.js";
-
-const styles = Styles;
 
 export default class RadioButton extends React.Component{
 
-  answerSelected(){
-    // AssessmentActions.answerSelected(this.props.item); TODO
+  static propTypes = {
+    item: React.PropTypes.object.isRequired,
```
Comment (fugue88) [inline client/js/components/common/radio_button.jsx]: Document.

File: client/js/components/common/radio_button.jsx [hunk 3]
```diff
@@ -2,22 +2,27 @@
...
 
   checkedStatus(){
-    var checked     = null;
-    var optionFlag  = null;
+    let checked = null;
+    let optionFlag = null;
 
-    if(this.props.checked === true) {
-      checked = "true";
+    if( this.props.checked === true ) {
```
Comment (fugue88) [inline client/js/components/common/radio_button.jsx]: Bad spacing.  Please do either `if(cond) {` or `if (cond) {`.

File: client/js/components/common/radio_button.jsx [hunk 4]
```diff
@@ -49,27 +52,17 @@ export default class RadioButton extends React.Component{
...
-        <div className="btn btn-block btn-question" style={Styles.btnQuestion}>
+        <div className="btn btn-block btn-question">
           <label>
             <input type="radio"
                    defaultChecked={this.checkedStatus()}
                    disabled={this.props.isDisabled}
                    name={this.props.name}
-                   onClick={() => { this.answerSelected(); }} />
-            <span style={Styles.span}>{this.props.item.material}</span>
+                   onClick={() => { this.selectAnswer(); }} />
```
Comment (fugue88) [inline client/js/components/common/radio_button.jsx]: Wonderful!

File: client/js/components/main/assessment.jsx [hunk 5]
```diff
@@ -94,24 +94,24 @@ export class Assessment extends React.Component{
...
-          messageIndex     = {props.progress.answerMessageIndex[index]}
-          allQuestions     = {props.allQuestions}
-          studentAnswers   = {{/*this.props.studentAnswers*/}}
-          outcomes         = {props.outcomes}
-          goToNextQuestion = {() => {props.nextQuestion();}}
-          goToPrevQuestion = {() => {props.previousQuestion();}}
-          submitAssessment = {() => {this.submitAssessment();}}
-      />
-    );
+    debugger;
```
Comment (fugue88) [inline client/js/components/main/assessment.jsx]: Oops!

File: client/js/components/main/assessment.jsx [hunk 6]
```diff
@@ -94,24 +94,24 @@ export class Assessment extends React.Component{
...
+    return <Item
```
Comment (fugue88) [inline client/js/components/main/assessment.jsx]: Put the parentheses back.  Joseph and Ben as well as I settled on it for several reasons.  Mine is that my editor then does nice auto-formatting.  I don't know their reasons.

File: client/js/components/main/assessment.jsx [hunk 7]
```diff
@@ -94,24 +94,24 @@ export class Assessment extends React.Component{
...
+      settings         = {props.settings}
```
Comment (fugue88) [inline client/js/components/main/assessment.jsx]: What happened to the `key` attribute?

File: client/js/components/main/assessment.jsx (hunk 6 above)
Comment (nick-benoit14) [inline client/js/components/main/assessment.jsx]: We should talk to Dave Spencer about this for the sake of consistency throughout the project. I removed the parentheses per his recommendation.

**Comment by fugue88** [comment]
Looks good.  Thanks for the changes.

**Pull request merged**